packages/space-opera/src/components/shared/card/card.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import '@material/mwc-button';
+
+import {customElement, html, LitElement, property} from 'lit-element';
+
+import {styles} from './styles.css.js';
+
+/**
+ * A styled checkbox with label
+ */
+@customElement('me-card')
+export class CardElement extends LitElement {
+  static styles = styles;
+  @property({type: String}) title = '';
+  @property({type: String}) functionId = '';
+  @property({type: String}) functionTitle = '';
+  @property({type: Function}) uploadFunction?: Function;
+  @property({type: Function}) copyFunction?: Function;
+  @property({type: Function}) removeFunction?: Function;
+  @property({type: Function}) undoFunction?: Function;
+
+  render() {
+    const upload = this.uploadFunction !== undefined ? html`
+    <mwc-button class="upload" id="uploadButton"
+      icon="cloud_upload" @click="${this.uploadFunction}">
+    </mwc-button>` :
+                                                       html``;
+    const copy = this.copyFunction !== undefined ? html`
+    <mwc-button class="upload"
+      icon="file_copy" @click="${this.copyFunction}">
+    </mwc-button>` :
+                                                   html``;
+    const remove = this.removeFunction !== undefined ? html`
+    <mwc-button class="upload" id="remove-hotspot"
+      icon="delete"
+      @click="${this.removeFunction}">
+    </mwc-button>
+    ` :
+                                                       html``;
+    const undo = this.undoFunction !== undefined ? html`
```

Comment:
Another place to replace undo with revert.

Reply:
done